tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,289 @@
+    node.exec_command(
+        sudo=True,
+        cmd=f"htpasswd -bBc {OPT_AUTH}/htpasswd \
+             {PRIVATE_REG_USER} {PRIVATE_REG_PASS}",
+    )
+
+    # Create a self-signed certificate
+    node.exec_command(
+        cmd=f'(echo "."; echo ".";echo ".";echo ".";echo ".";\
+                echo "{node.hostname}";echo ".";)| sudo openssl req -newkey \
+                rsa:4096 -nodes -sha256 -keyout {OPT_CERTS}/domain.key \
+                -x509 -days 365 -out {OPT_CERTS}/domain.crt -addext \
+                "subjectAltName = DNS:{node.hostname}"'
+    )
+
+    # Create a symbolic link to domain.cert
+    node.exec_command(
+        sudo=True,
+        cmd=f"ln -s {OPT_CERTS}/domain.crt \
+            {OPT_CERTS}/domain.cert",
+    )
+
+    # Add certificate to trusted list on the private registry node
+    node.exec_command(
+        sudo=True,
+        cmd=f"cp {OPT_CERTS}/domain.crt \
+            /etc/pki/ca-trust/source/anchors/",
+    )
+    node.exec_command(sudo=True, cmd="update-ca-trust")
+    update_cert(node)
+
+    # Perform passwordless ssh between all nodes
+    ceph_cluster.setup_ssh_keys()
+    file_name = f"{OPT_CERTS}/domain.crt"
+
+    # Copy the certificate to other nodes
+    for ceph in ceph_nodes:
+        ceph.exec_command(
+            sudo=True, cmd="touch /etc/pki/ca-trust/source/anchors/domain.crt"
+        )
+        contents, _ = node.exec_command(sudo=True, cmd=f"cat {file_name}")
+        key_file = ceph.remote_file(
+            sudo=True,
+            file_name="/etc/pki/ca-trust/source/anchors/domain.crt",
+            file_mode="a",
+        )
+        key_file.write(contents)
+        key_file.flush()
+        ceph.exec_command(sudo=True, cmd="update-ca-trust")
+        update_cert(node)
+
+    # Login to registry.redhat.io
+    con = get_cephci_config()
+    username_ = con.get("cdn_credentials", {}).get("username")
+    password_ = con.get("cdn_credentials", {}).get("password")
+    for ceph in ceph_nodes:
+        ceph.exec_command(
+            sudo=True,
+            cmd=f"podman login -u {username_} -p {password_} registry.redhat.io",
+        )
+
+    # Start Private registry on Node
+    start_private_registry(node)
+
+    # Add images to private registry
+    private_registry_add_image(node, config)
+
+    # Login to private registy on all nodes
+    for ceph in ceph_nodes:
+        ceph.exec_command(
+            sudo=True,
+            cmd=f"podman login -u {PRIVATE_REG_USER} -p {PRIVATE_REG_PASS} {node.hostname}:5000",
+        )
+
+    # Pull the images to the nodes
+    out = loads(
+        node.exec_command(
+            sudo=True,
+            cmd=f"curl -u {PRIVATE_REG_USER}:{PRIVATE_REG_PASS} https://{node.hostname}:5000/v2/_catalog",
+        )[0]
+    )
+    repos = out.get("repositories")
+    for ceph in ceph_nodes:
+        for img in repos:
+            ceph.exec_command(sudo=True, cmd=f"podman pull {node.hostname}:5000/{img}")
```

Comment:
move this to - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py